ibm/resource_ibm_iam_authorization_policy.go

```diff
@@ -112,137 +113,167 @@ func resourceIBMIAMAuthorizationPolicyCreate(d *schema.ResourceData, meta interf
+			Name:  core.StringPtr("resourceType"),
+			Value: core.StringPtr(sType.(string)),
+		}
+		policySubject.Attributes = append(policySubject.Attributes, resourceTypeSubjectAttribute)
 	}
 
 	if tType, ok := d.GetOk("target_resource_type"); ok {
-		policy.Resources[0].SetResourceType(tType.(string))
+		resourceTypeResourceAttribute := iampolicymanagementv1.ResourceAttribute{
+			Name:  core.StringPtr("resourceType"),
+			Value: core.StringPtr(tType.(string)),
+		}
+		policyResource.Attributes = append(policyResource.Attributes, resourceTypeResourceAttribute)
 	}
 
 	if sResGrpID, ok := d.GetOk("source_resource_group_id"); ok {
-		policy.Subjects[0].SetResourceGroupID(sResGrpID.(string))
+		resourceGroupSubjectAttribute := iampolicymanagementv1.SubjectAttribute{
+			Name:  core.StringPtr("resourceGroupId"),
+			Value: core.StringPtr(sResGrpID.(string)),
+		}
+		policySubject.Attributes = append(policySubject.Attributes, resourceGroupSubjectAttribute)
 	}
 
 	if tResGrpID, ok := d.GetOk("target_resource_group_id"); ok {
-		policy.Resources[0].SetResourceGroupID(tResGrpID.(string))
+		resourceGroupResourceAttribute := iampolicymanagementv1.ResourceAttribute{
+			Name:  core.StringPtr("resourceGroupId"),
+			Value: core.StringPtr(tResGrpID.(string)),
+		}
+		policyResource.Attributes = append(policyResource.Attributes, resourceGroupResourceAttribute)
 	}
 
 	roles, err := getAuthorizationRolesByName(expandStringList(d.Get("roles").([]interface{})), sourceServiceName, targetServiceName, meta)
 	if err != nil {
 		return err
 	}
 
-	policy.Roles = iampapv1.ConvertRoleModels(roles)
-
-	authPolicy, err := iampapClient.V1Policy().Create(policy)
+	createPolicyOptions := iampapClient.NewCreatePolicyOptions(
+		"authorization",
+		[]iampolicymanagementv1.PolicySubject{*policySubject},
+		roles,
+		[]iampolicymanagementv1.PolicyResource{*policyResource},
+	)
+	authPolicy, _, err := iampapClient.CreatePolicy(createPolicyOptions)
 
 	if err != nil {
 		return fmt.Errorf("Error creating authorization policy: %s", err)
 	}
 
-	d.SetId(authPolicy.ID)
+	d.SetId(*authPolicy.ID)
 
 	return resourceIBMIAMAuthorizationPolicyRead(d, meta)
 }
 
 func resourceIBMIAMAuthorizationPolicyRead(d *schema.ResourceData, meta interface{}) error {
 
-	iampapClient, err := meta.(ClientSession).IAMPAPAPI()
+	iampapClient, err := meta.(ClientSession).IAMPolicyManagementV1API()
 	if err != nil {
 		return err
 	}
 
-	authorizationPolicy, err := iampapClient.V1Policy().Get(d.Id())
+	getPolicyOptions := &iampolicymanagementv1.GetPolicyOptions{
+		PolicyID: core.StringPtr(d.Id()),
+	}
+
+	authorizationPolicy, _, err := iampapClient.GetPolicy(getPolicyOptions)
 	if err != nil {
 		return fmt.Errorf("Error retrieving authorizationPolicy: %s", err)
 	}
 	roles := make([]string, len(authorizationPolicy.Roles))
 	for i, role := range authorizationPolicy.Roles {
-		roles[i] = role.Name
+		roles[i] = *role.DisplayName
 	}
 	d.Set("roles", roles)
-	d.Set("version", authorizationPolicy.Version)
```

Comment:
The reason it is not set is because:
1.  We are not storing the ETag as a value for the `version` key given authorizathion policy does not include an Update implementation (ex. returns `nil`) same as in IBM CLI
2. For other type of policies like userpolicy, we get the ETag header by doing a GET v1/policy right before we do a PUT. 
https://github.com/guicabr/terraform-provider-ibm/blob/2ec549e976d5c9d84c4079adb95dcd159f19966b/ibm/resource_ibm_iam_user_policy.go#L312

@swcolley correct me if I'm wrong 🤠